main.go

```diff
@@ -27,8 +27,10 @@ func (Factory) Validate(m *plugins.Manager, config []byte) (interface{}, error)
 	return internal.Validate(m, config)
 }
 
-func init() {
+// Init register plugin factory with OPA.
+func Init() error {
 	runtime.RegisterPlugin("envoy.ext_authz.grpc", Factory{})
+	return nil
 }
 
 func main() {
```

Comment:
It's a bit odd that we still have the `main` function since we're not changing the entry point behaviour at all. I'm guessing this is due to the fact the plugin build uses the vendored OPA which is incompatible with non-vendored OPA packages in built OPA executables? This is more of a question than a comment. Is it possible to get rid of the `main` function? If not, a comment explaining why we need it would be useful.

Reply:
Added a comment.